internal/featuretests/v3/upstreamtls_test.go

```diff
@@ -340,3 +340,119 @@ func TestUpstreamTLSWithHTTPRoute(t *testing.T) {
 		TypeUrl: clusterType,
 	})
 }
+
+func TestBackendTLSPolicyPrecedenceOverUpstreamProtocolAnnotationWithHTTPRoute(t *testing.T) {
+	rh, c, done := setup(t, func(b *dag.Builder) {
+		for _, processor := range b.Processors {
+			if gatewayAPIProcessor, ok := processor.(*dag.GatewayAPIProcessor); ok {
+				gatewayAPIProcessor.UpstreamTLS = &dag.UpstreamTLS{
+					MinimumProtocolVersion: "1.2",
+					MaximumProtocolVersion: "1.2",
+				}
+			}
+		}
+	})
+	defer done()
+
+	sec1 := featuretests.TLSSecret(t, "sec1", &featuretests.ClientCertificate)
```

Comment:
last nit, otherwise LGTM, I think we end up actually only using one of these secrets (`sec2`)? could remove and consolidate to one